src/blog/2024/03/http-authentication-node-red-with-flowfuse.md

```diff
@@ -0,0 +1,37 @@
+---
+title: HTTP Node Authentication in Node-RED with FlowFuse
+subtitle: Choosing the right form of authentication for your Node-RED integration is important.
+description: In this blog we will discuss the various forms of authentication for integrating web applications with Node-RED.
+date: 2024-03-25
+authors: ["grey-dziuba"]
+image: 
+tags:
+    - posts
+    - flowfuse
+    - http in
+    - http node
+    - api
+---
+
+In today’s interconnected world, web development often involves interacting with APIs through a variety of authentication methods for secure data exchange. As a citizen developer using Node-RED and FlowFuse, it's essential to understand these authentication types and how they can be utilized in your projects.
+
+<!--more-->
+
+
+## Basic Authentication: A Simple Approach
+
+Basic Authentication is widely supported and straightforward to implement. It requires users to provide a username and password before accessing the Node-RED instance. In FlowFuse, developers can create application-specific usernames and passwords for Basic Authentication, making it suitable for internal applications or environments with tight control over access. Remember that the security of this method relies on secure connections like HTTPS to prevent credential interception.
+
+## FlowFuse Authentication: Seamless Integration
+
+FlowFuse Authentication is similar to Basic Authentication, with one distinct difference. It allows users to authenticate using the same credentials as their FlowFuse platform login. This approach is excellent for when a citizen developer wants to create an application backend that will only be used by FlowFuse-authenticated team members, outside of that it functions the same as the Basic Authentication method.  This prevents the need to share passwords between team members because they will use their own authentication to interact with the application.
```

Comment:
It's not similar, very different. HTTP Basic Auth is very easy to use for APIs as you can write `https://username:password@your-instance.flowfuse.cloud` and it works. For automated access, not person bound, this method is hard to use and you need PATs. (You've not mentioned Personal access tokens, can you add that?)

Reply:
```suggestion
FlowFuse authentication offers a seamless and secure way for users to access dashboards and other resources that are typically accessed through a browser. It leverages single sign-on (SSO) and SAML 2.0, reducing the management burden for organizations. 

For users this is convenient as they can access multiple applications and resources using a single set of credentials, eliminating the need to remember and manage multiple passwords.

For organizations, SSO enhances security by centralizing authentication and authorization, reducing the risk of unauthorized access. By leveraging SSO and SAML 2.0, FlowFuse takes care of user management, freeing up customers from the administrative burden of managing user accounts and passwords. FlowFuse authentication adheres to industry-standard security protocols, ensuring compliance with regulatory requirements.

This method of authentication is however impractical for API access by other services, to programmatically transfer data between them.

```